File: backend/src/main/java/com/ucto/backend/dto/SubscriptionPlan.java
package com.ucto.backend.dto;

public class SubscriptionPlan {
    private String tier;
    private int price;         // Price in INR paise (e.g., 2999 = ₹2,999)
    private int maxProjects;
    private int maxAgentRuns;
    private boolean hasAudit;
    private boolean hasCompliance;
    private boolean hasPrioritySupport;

    public SubscriptionPlan() {}

    public SubscriptionPlan(String tier, int price, int maxProjects, int maxAgentRuns,
                            boolean hasAudit, boolean hasCompliance, boolean hasPrioritySupport) {
        this.tier = tier;
        this.price = price;
        this.maxProjects = maxProjects;
        this.maxAgentRuns = maxAgentRuns;
        this.hasAudit = hasAudit;
        this.hasCompliance = hasCompliance;
        this.hasPrioritySupport = hasPrioritySupport;
    }

    public String getTier() { return tier; }
    public void setTier(String tier) { this.tier = tier; }

    public int getPrice() { return price; }
    public void setPrice(int price) { this.price = price; }

    public int getMaxProjects() { return maxProjects; }
    public void setMaxProjects(int maxProjects) { this.maxProjects = maxProjects; }

    public int getMaxAgentRuns() { return maxAgentRuns; }
    public void setMaxAgentRuns(int maxAgentRuns) { this.maxAgentRuns = maxAgentRuns; }

    public boolean isHasAudit() { return hasAudit; }
    public void setHasAudit(boolean hasAudit) { this.hasAudit = hasAudit; }

    public boolean isHasCompliance() { return hasCompliance; }
    public void setHasCompliance(boolean hasCompliance) { this.hasCompliance = hasCompliance; }

    public boolean isHasPrioritySupport() { return hasPrioritySupport; }
    public void setHasPrioritySupport(boolean hasPrioritySupport) { this.hasPrioritySupport = hasPrioritySupport; }

    public String getFormattedPrice() {
        if (price == 0) return "Free";
        return "₹" + String.format("%,d", price);
    }

    public String getCurrency() { return "INR"; }
}

File: frontend/lib/models/subscription_plan.dart
class SubscriptionPlan {
  final String tier;
  final int price;
  final int maxProjects;
  final int maxAgentRuns;
  final bool hasAudit;
  final bool hasCompliance;
  final bool hasPrioritySupport;

  SubscriptionPlan({
    required this.tier,
    required this.price,
    required this.maxProjects,
    required this.maxAgentRuns,
    required this.hasAudit,
    required this.hasCompliance,
    required this.hasPrioritySupport,
  });

  factory SubscriptionPlan.fromJson(Map<String, dynamic> json) {
    return SubscriptionPlan(
      tier: json['tier'] as String,
      price: json['price'] as int,
      maxProjects: json['maxProjects'] as int,
      maxAgentRuns: json['maxAgentRuns'] as int,
      hasAudit: json['hasAudit'] as bool? ?? false,
      hasCompliance: json['hasCompliance'] as bool? ?? false,
      hasPrioritySupport: json['hasPrioritySupport'] as bool? ?? false,
    );
  }

  String get formattedPrice {
    if (price == 0) return 'Free';
    return '₹${price ~/ 100},${price % 100 == 0 ? (price % 100).toString().padLeft(2, '0') : (price % 100).toString()}';
  }

  String get formattedPriceUSD {
    if (price == 0) return '\$0';
    final usd = (price / 85).round();
    return '\$$usd';
  }
}
